payu/models/cice.py

```diff
@@ -62,27 +62,63 @@ def set_model_pathnames(self):
         setup_nml = self.ice_in['setup_nml']
 
         res_path = os.path.normpath(setup_nml['restart_dir'])
+        input_dir = setup_nml.get('input_dir', None)
+
+        if input_dir is None:
+            # Default to reading and writing inputs/restarts in-place
+            input_path = res_path
+        else:
+            input_path = os.path.normpath(input_dir)
```

Comment:
Should this be absolute path?  (Genuinely not sure BTW)

Reply:
It is kept a relative path until the end (line 108) where it is joined to `work_path`. It is done like that so it can be compared with the other paths on the grid, kmt and restart pointer files.